File: java/tutorial/tut5-shoot_kill_ai/src/infinity/gameobjectproperties/AiIdentifiableObjectInfo.java
package infinity.gameobjectproperties;

import infinity.gameengine.Vector2;

public class AiIdentifiableObjectInfo {

	public int GameObjectId;
	public Vector2 Position;
	public GameObjectType GameObjectType;
	public Vector2 Velocity;

    public void Create(int gameObjectId, GameObjectType gameObjectType, Vector2 position, Vector2 velocity)
    {
        GameObjectType = gameObjectType;
        GameObjectId = gameObjectId;
        Position = position;
        Velocity = velocity;
    }

}

File: src/Infinity/Infinity/AiIdentifiableObjectInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework;

namespace Infinity
{
    public class AiIdentifiableObjectInfo
    {
        public GameObjectType GameObjectType { get; set; }
        public int GameObjectId { get; set; }
        public Vector2 Position { get; set; }
        public Vector2 Velocity { get; set; }

        public void Create(int gameObjectId, GameObjectType gameObjectType, Vector2 position, Vector2 velocity)
        {
            GameObjectType = gameObjectType;
            GameObjectId = gameObjectId;
            Position = position;
            Velocity = velocity;
        }
    }

    public enum GameObjectType
    {
        Spacecraft,
        Bullet
    }
}
